icicle/include/icicle/sumcheck/sumcheck_proof.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+#include <vector>
+#include <memory>
+
+namespace icicle {
+
+  /**
+   * @brief Represents a sumcheck proof.
+   *
+   * This class encapsulates the sumcheck proof, contains the evaluations of the round polynomials
+   * for each layer at the sumcheck proof.
+   * Evaluations are at x = 0, 1, 2 ... K
+   * Where K is the degree of the combiine function used at the sumcheck protocol.
+   *
+   * @tparam S Type of the field element (e.g., prime field or extension field elements).
+   */
+
+  template <typename S>
+  class SumCheckProof
+  {
+  public:
+    // Constructor
+    SumCheckProof(uint nof_round_polynomials, uint round_polynomial_degree)
```

Comment:
what is uint? probably defined in some headers but better stick to standard uint32_t or similar